tests/sentry/api/endpoints/test_organization_details.py

```diff
@@ -783,6 +787,84 @@ def test_org_mapping_already_taken(self):
         OrganizationMapping.objects.create(organization_id=999, slug="taken", region_name="us")
         self.get_error_response(self.organization.slug, slug="taken", status_code=409)
 
+    def test_configure_auth_provider(self):
+        with self.feature("organizations:api-auth-provider"):
+            old_config = {"domain": "foo.com"}
+            new_config = {"domain": "bar.com"}
+            provider = "google"
+
+            self.get_success_response(
+                self.organization.slug,
+                method="put",
+                providerKey=provider,
+                providerConfig=old_config,
+            )
+            auth_provider = AuthProvider.objects.get(organization_id=self.organization.id)
+            assert auth_provider.provider == provider
+            assert auth_provider.config == {
+                "domains": ["foo.com"],
+                "version": DATA_VERSION,
+                "sentry-source": "api-organization-details",
+            }
+
+            self.get_success_response(
+                self.organization.slug,
+                method="put",
+                providerKey=provider,
+                providerConfig=new_config,
+            )
+            auth_provider = AuthProvider.objects.get(organization_id=self.organization.id)
+            assert auth_provider.provider == provider
+            assert auth_provider.config == {
+                "domains": ["bar.com"],
+                "version": DATA_VERSION,
+                "sentry-source": "api-organization-details",
+            }
+
+    def test_invalid_auth_provider_configuration(self):
+        with self.feature("organizations:api-auth-provider"):
+            self.get_error_response(
+                self.organization.slug, method="put", providerKey="google", status_code=400
+            )
+            self.get_error_response(
+                self.organization.slug,
+                method="put",
+                providerConfig={"domain": "foo.com"},
+                status_code=400,
+            )
+            self.get_error_response(
+                self.organization.slug,
+                method="put",
+                providerKey="not_valid",
+                providerConfig={"domain": "foo.com"},
+                status_code=400,
+            )
+
+            response = self.get_error_response(
+                self.organization.slug,
+                method="put",
+                providerKey="google",
+                providerConfig={"invalid_domain": "foo.com"},
+                status_code=400,
+            )
+            assert response.data == {
+                "providerConfig": ["Invalid providerConfig for authprovider google"]
```

Comment:
Going back to the KeyError. If I received this error message, I would be frustrated as an end-user because I don't even know where to begin to fix it.

Reply:
See above